webapp/components/signup/signup_controller.jsx

```diff
@@ -0,0 +1,322 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+
+import FormError from 'components/form_error.jsx';
+import LoadingScreen from 'components/loading_screen.jsx';
+
+import UserStore from 'stores/user_store.jsx';
+import BrowserStore from 'stores/browser_store.jsx';
+
+import * as AsyncClient from 'utils/async_client.jsx';
+import Client from 'client/web_client.jsx';
+import * as GlobalActions from 'actions/global_actions.jsx';
+
+import logoImage from 'images/logo.png';
+import ErrorBar from 'components/error_bar.jsx';
+
+import {FormattedMessage} from 'react-intl';
+import {browserHistory, Link} from 'react-router/es6';
+
+export default class SignupController extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            emailEnabled: global.window.mm_config.EnableSignUpWithEmail === 'true',
+            gitlabEnabled: global.window.mm_config.EnableSignUpWithGitLab === 'true',
+            googleEnabled: global.window.mm_config.EnableSignUpWithGoogle === 'true',
+            office365Enabled: global.window.mm_config.EnableSignUpWithOffice365 === 'true',
+            ldapEnabled: global.window.mm_license.IsLicensed === 'true' && global.window.mm_config.EnableLdap === 'true',
+            samlEnabled: global.window.mm_license.IsLicensed === 'true' && global.window.mm_config.EnableSaml === 'true',
+            teamName: '',
+            serverError: '',
+            noOpenServerError: false,
+            loading: true
+        };
+    }
+
+    componentWillMount() {
```

Comment:
This was from `signup_user_complete`, if the user is already logged in but not in the invited team, add the user to the team. Otherwise, ensure that the invitation is valid.